lib/spack/llnl/util/filesystem.py

```diff
@@ -937,9 +945,8 @@ def is_nonsymlink_exe_with_shebang(path):
             return False
 
         # Should start with a shebang
-        with open(path, "rb") as f:
-            return f.read(2) == b"#!"
-    except (IOError, OSError):
+        return has_shebang(path)
```

Comment:
(nitpick) Since the function has its own try/except block, can we extract it from _this_ try/except?

Reply:
I think there's a misunderstanding. What I'm saying is:
```python
def is_nonsymlink_exe_with_shebang(path):
    try:
        st = os.lstat(path)
        # Should not be a symlink
        if stat.S_ISLNK(st.st_mode):
            return False
        # Should be executable
        if not st.st_mode & (stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH):
            return False
    except OSError:
        return False

    return has_shebang(path)
```

Reply:
Oh